decision-records/2024-11-12--up-alert-rules.md

```diff
@@ -0,0 +1,64 @@
+# Up Alert Rules
+**Date:** 2024-11-12
+**Authors:** @dstathis
+
+## Context and Problem Statement
+
+Charms in the COS ecosystem currently have a wide range of different alert rules pertaining to the "up" metric. This
+particular metric exists for all scrape targets and always behaves the same. Therefore we should have a uniform set of
+alert rules.
+
+## Accepted Solution
+
+The folowing alert rule group should be used in each charm:
+
+```
+groups:
+- name: HostHealth
+  rules:
+  - alert: HostUnreachable
+    expr: up < 1
+    for: 0m
+    labels:
+      severity: warning
+    annotations:
+      summary: Host '{{ $labels.instance }}' is unreachable.
+      description: >-
+        Host '{{ $labels.instance }}' is unreachable. This could indicate an availability issue and should be investigated.
+  - alert: HostDown
+    expr: up < 1
+    for: 5m
+    labels:
+      severity: critical
+    annotations:
+      summary: Host '{{ $labels.instance }}' is down.
+      description: >-
+        Host '{{ $labels.instance }}' is down.
+  - alert: HostMetricsMissing
+    expr: absent(up)
+    for: 5m
+    labels:
+      severity: critical
+    annotations:
+      summary: Metrics not received from host '{{ $labels.instance }}'.
+      description: >-
+        Metrics not received from host '{{ $labels.instance }}'. This is likely due to an issue with an itermediate metrics agent.
+```
+
+## Rationale
```

Comment:
In the rationale we should explicitly address:
- Why per charm and not just once in prometheus / mimir?
- Should `up` alert be in prom and `absent` per charm? 
- How should aggregators such as gagent go about this. See also `up < 1 or absent(up)` in https://github.com/canonical/grafana-agent-operator/pull/148.
- Scrape failed is different from host didn't send data.
- Missing of scrape targets vs missing of self, see https://github.com/canonical/prometheus-k8s-operator/blob/main/src/prometheus_alert_rules/prometheus_missing.rule